llvm/include/llvm/ADT/EquivalenceClasses.h

```diff
@@ -220,6 +220,46 @@ template <class ElemTy> class EquivalenceClasses {
     return *ECV;
   }
 
+  /// erase - Erase a value from the union/find set, return if erase succeed.
+  bool erase(const ElemTy &V) {
+    if (!TheMapping.contains(V))
+      return false;
+    const ECValue *Cur = TheMapping[V];
+    const ECValue *Next = Cur->getNext();
+    if (Cur->isLeader()) {
+      if (Next) {
+        Next->Leader = Cur->Leader;
+        Next->Next = (const ECValue *)((intptr_t)Next->Next | (intptr_t)1);
+        const ECValue *newLeader = Next;
+        while ((Next = Next->getNext())) {
+          Next->Leader = newLeader;
+        }
+      }
+    } else {
+      const ECValue *Leader = findLeader(V).Node;
+      const ECValue *Pre = Leader;
+      while (Pre->getNext() != Cur) {
+        Pre = Pre->getNext();
+      }
+      if (!Next) {
+        Pre->Next = nullptr;
+        Leader->Leader = Pre;
+      } else {
+        Pre->Next =
+            (const ECValue *)((intptr_t)Next | (intptr_t)Pre->isLeader());
+        Next->Leader = Pre;
+      }
+    }
+    TheMapping.erase(V);
+    for (auto I = Members.begin(); I != Members.end(); I++) {
```

Comment:
`Members.erase(llvm::find(...))`. And assert that the element was found, just to be sure...